lib/daru/maths/statistics/vector.rb

```diff
@@ -565,6 +565,34 @@ def emv(n = 10, wilder = false)
           Daru::Vector.new(var_base, index: @index, name: @name)	
         end
 
+        # Exponential Moving Standard Deviation.
+        # Calculates an exponential moving standard deviation of the series using a
+        # specified parameter. If wilder is false (the default) then the EMSD
+        # uses a smoothing value of 2 / (n + 1), if it is true then it uses the
+        # Welles Wilder smoother of 1 / n.
+        #
+        # @param [Integer] n (10) Loopback length.
+        # @param [TrueClass, FalseClass] wilder (false) If true, 1/n value is 
+        #   used for smoothing; if false, uses 2/(n+1) value
+        #
+        # @example Using emsd
+        #
+        #   ts = Daru::Vector.new((1..100).map { rand })
+        #            # => [0.400..., 0.727..., 0.862..., 0.013..., ...]
+        #
+        #   # first 9 observations are nil
+        #   ts.emsd   # => [ ... nil, 0.285... , 0.258..., 0.243..., ...]
+        #
+        # @return [Daru::Vector] contains EMSD
+        def emsd(n = 10, wilder = false)
+          result = []
+          vec = emv(n, wilder)
+          vec.data.each do |d|
+            result << (@missing_values.has_key?(d) ? nil : Math::sqrt(d))
```

Comment:
You have a vector in `vec` and you're using the `@missing_values` parameter of `self`. Use missing values of `vec` directly.

Also, `vec` is a very random name. Call it something specific.